%% Generated by Sphinx.
\def\sphinxdocclass{article}
\documentclass[letterpaper,10pt,english]{sphinxhowto}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{times}
\expandafter\ifx\csname T@LGR\endcsname\relax
\else
% LGR was declared as font encoding
  \substitutefont{LGR}{\rmdefault}{cmr}
  \substitutefont{LGR}{\sfdefault}{cmss}
  \substitutefont{LGR}{\ttdefault}{cmtt}
\fi
\expandafter\ifx\csname T@X2\endcsname\relax
  \expandafter\ifx\csname T@T2A\endcsname\relax
  \else
  % T2A was declared as font encoding
    \substitutefont{T2A}{\rmdefault}{cmr}
    \substitutefont{T2A}{\sfdefault}{cmss}
    \substitutefont{T2A}{\ttdefault}{cmtt}
  \fi
\else
% X2 was declared as font encoding
  \substitutefont{X2}{\rmdefault}{cmr}
  \substitutefont{X2}{\sfdefault}{cmss}
  \substitutefont{X2}{\ttdefault}{cmtt}
\fi


\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=\small}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}


\usepackage{sphinxmessages}
\setcounter{tocdepth}{4}
\setcounter{secnumdepth}{4}


\title{CNVkit}
\date{Apr 22, 2021}
\release{0.1.0}
\author{Anibal Morales (PBGL)}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Version 0.1.0}
\makeindex
\begin{document}

\pagestyle{empty}

        \pagenumbering{Roman} %%% to avoid page 1 conflict with actual page
        \begin{titlepage}
            \vspace*{10mm} %%% * is used to give space from top
            \flushright\textbf{\Huge {PBGL CNVkit Analysis v1.0\\}}
            \vspace{0mm} %%% * is used to give space from top
            \textbf{\Large {A Laboratory Manual\\}}
            \vspace{50mm}
            \textbf{\Large {Anibal E. Morales-Zambrana\\}}
            \vspace{10mm}
            \textbf{\Large {Plant Breeding and Genetics Laboratory\\}}
            \vspace{0mm}
            \textbf{\Large {FAO/IAEA Joint Division\\}}
            \vspace{0mm}
            \textbf{\Large {Seibersdorf, Austria\\}}
	    \vspace{10mm}
            \normalsize Created: April, 2021\\
            \vspace*{0mm}
            \normalsize  Last updated: 21 April 2021
            %% \vfill adds at the bottom
            \vfill
            \small\flushleft {{\textbf {Please note:}} \textit {This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views
expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.}}
        \end{titlepage}
        \pagenumbering{arabic}
        \newcommand{\sectionbreak}{\clearpage}

\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}



\section{Background}
\label{\detokenize{index:background}}
\sphinxAtStartPar
\sphinxstylestrong{{[}DRAFT{]}}

\sphinxAtStartPar
Copy number variation (CNV) analysis using CNVkit, R, Jupyter Notebooks, Miniconda3, Git, along other packages.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
This is not an official IAEA publication but is made available as working material. The material has not undergone an official review by the IAEA. The views expressed do not necessarily reflect those of the International Atomic Energy Agency or its Member States and remain the responsibility of the contributors. The use of particular designations of countries or territories does not imply any judgement by the publisher, the IAEA, as to the legal status of such countries or territories, of their authorities and institutions or of the delimitation of their boundaries. The mention of names of specific companies or products (whether or not indicated as registered) does not imply any intention to infringe proprietary rights, nor should it be construed as an endorsement or recommendation on the part of the IAEA.
\end{sphinxadmonition}


\section{Installations}
\label{\detokenize{index:installations}}
\sphinxAtStartPar
Before installing any necessary software, it is recommended to check if the computer is running 32\sphinxhyphen{}bit or 64\sphinxhyphen{}bit for downloading Miniconda3. Run the following to verify the system:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} uname \PYGZhy{}m
\end{sphinxVerbatim}


\subsection{Miniconda3 (conda)}
\label{\detokenize{index:miniconda3-conda}}
\sphinxAtStartPar
Download the Miniconda3, or simply “conda”, installer:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://docs.conda.io/en/latest/miniconda.html\#linux-installers}{Miniconda3 installer for Linux}

\end{itemize}

\sphinxAtStartPar
Run the downloaded installer (for a 64\sphinxhyphen{}bit system):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} bash Miniconda3\PYGZhy{}latest\PYGZhy{}Linux\PYGZhy{}x86\PYGZus{}64.sh
\end{sphinxVerbatim}

\sphinxAtStartPar
Open a new terminal window for conda to take effect. Verify the installation in new terminal window and update conda:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\PYGZdl{} conda update \PYGZhy{}\PYGZhy{}all
\PYGZdl{} conda upgrade \PYGZhy{}\PYGZhy{}all
\end{sphinxVerbatim}


\subsection{Git with conda}
\label{\detokenize{index:git-with-conda}}
\sphinxAtStartPar
Git will be installed first to clone locally (download a copy to your local computer) the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} repository from GitHub. To do so, run the following:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install \PYGZhy{}c anaconda git
\end{sphinxVerbatim}


\subsubsection{Cloning the Necessary Repositories}
\label{\detokenize{index:cloning-the-necessary-repositories}}
\sphinxAtStartPar
After the installation, clone the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} repository to the local computer in the desired directory.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone https://github.com/amora197/pbgl\PYGZhy{}cnvkit.git
\end{sphinxVerbatim}

\sphinxAtStartPar
A folder called \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} should be listed in the directory. Navigate into it and inspect its items.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} cd pbgl\PYGZhy{}cnvkit
\PYGZdl{} ls \PYGZhy{}l
\end{sphinxVerbatim}

\sphinxAtStartPar
The \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} directory should contain:
\begin{itemize}
\item {} 
\sphinxAtStartPar
3 folders:
\begin{itemize}
\item {} 
\sphinxAtStartPar
docs

\item {} 
\sphinxAtStartPar
envs

\item {} 
\sphinxAtStartPar
output

\end{itemize}

\item {} 
\sphinxAtStartPar
3 files:
\begin{itemize}
\item {} 
\sphinxAtStartPar
cnvkit\sphinxhyphen{}analysis.ipynb

\item {} 
\sphinxAtStartPar
config\sphinxhyphen{}cnvkit.yml

\item {} 
\sphinxAtStartPar
README.rst

\end{itemize}

\end{itemize}

\sphinxAtStartPar
Once inside the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} directory, clone \sphinxstylestrong{etal/cnvkit} repository that contains the workflow and source code for analyzing copy number variations/alterations.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} git clone \PYGZhy{}\PYGZhy{}branch v0.9.7 \PYGZhy{}\PYGZhy{}single\PYGZhy{}branch https://github.com/etal/cnvkit.git
\PYGZdl{} ls \PYGZhy{}l
\end{sphinxVerbatim}

\sphinxAtStartPar
A new directory \sphinxstylestrong{cnvkit} should be present.


\subsection{Required Libraries with conda}
\label{\detokenize{index:required-libraries-with-conda}}
\sphinxAtStartPar
cnvkit has multiple dependencies, listed below:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Git

\item {} 
\sphinxAtStartPar
cnvkit

\item {} 
\sphinxAtStartPar
Jupyter Notebook

\end{itemize}

\sphinxAtStartPar
There are two ways to install the rest of the necessary libraries to run cnvkit: automatically or manually. The former is slower, providing a long coffee break (sometimes overnight durations) while the conda installations run. The latter proves a faster way to get the tool up\sphinxhyphen{}and\sphinxhyphen{}running.


\subsubsection{Automatically (slower)}
\label{\detokenize{index:automatically-slower}}
\sphinxAtStartPar
One YAML file, \sphinxstylestrong{environment.yml}, is provided inside the \sphinxstylestrong{envs/} directory to automatically create a conda environment and install the dependent libraries. This creates the conda environment, along all necessary packages to run cnvkit. Run \sphinxstylestrong{environment.yml}:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env create \PYGZhy{}\PYGZhy{}file envs/environment.yml
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, the created environment can be verified running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env list
\end{sphinxVerbatim}

\sphinxAtStartPar
Activate the created environment (\sphinxstylestrong{cnvkit}):

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate cnvkit
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\subsubsection{Manually (faster)}
\label{\detokenize{index:manually-faster}}
\sphinxAtStartPar
To manually create and activate an environment, run:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda create \PYGZhy{}\PYGZhy{}name cnvkit
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, the created environment can be verified running:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda env list
\end{sphinxVerbatim}

\sphinxAtStartPar
Activate the virtual environment with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda activate cnvkit
\end{sphinxVerbatim}

\sphinxAtStartPar
Start running the installations of the necessary libraries, paying attention to the prompts for each one:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda install pyyaml
\PYGZdl{} conda install cnvkit
\PYGZdl{} conda install notebook
\end{sphinxVerbatim}

\sphinxAtStartPar
Once done, all the necessary packages should be installed. This can be verified with:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} conda list
\end{sphinxVerbatim}


\section{Running a Jupyter Notebook}
\label{\detokenize{index:running-a-jupyter-notebook}}
\sphinxAtStartPar
To access the Jupyter Notebooks, run the following command inside the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit} directory:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYGZdl{} jupyter notebook
\end{sphinxVerbatim}

\sphinxAtStartPar
This command will start a Jupyter Notebook session inside the directory the command is run. The user can navigate between directories, visualize files, and edit files in the browser by clicking on directories or files, respectively.

\sphinxAtStartPar
Look for \sphinxstylestrong{cnvkit\sphinxhyphen{}analysis.ipynb} and click on it to open the Jupyter Notebook and run the analysis.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
Jupyter lets the user duplicate, rename, move, download, view, or edit files in a web browser. This can be done by clicking the box next to a file and choosing accordingly.
\end{sphinxadmonition}


\section{Editing the Configuration File}
\label{\detokenize{index:editing-the-configuration-file}}
\sphinxAtStartPar
In order to run the CNVkit Jupyter Notebook, the user needs to feed it with a configuration file (\sphinxstylestrong{config\sphinxhyphen{}cnvkit.yml}) that specifies the paths to the bam files, comparisons to be done, chromosomes to analyze, and parameter definitions for calculating and plotting CNVs.

\sphinxAtStartPar
The configuration file \sphinxstylestrong{config\sphinxhyphen{}cnvkit.yml} can be found in the same directory as the Jupyter Notebook.

\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
The user needs to edit \sphinxstylestrong{config\sphinxhyphen{}cnvkit.yml} to point towards bam/bed/fasta files; specify comparisons and chromosomes to analyze; and define the output path.
\end{sphinxadmonition}

\sphinxAtStartPar
The configuration file \sphinxstylestrong{config\sphinxhyphen{}cnvkit.yml} contains multiple parameters to be defined by the user:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{paths}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
sample names and their respective paths to \sphinxstylestrong{.bam} files

\item {} 
\sphinxAtStartPar
samples can be named as desired but the sample name must be repeated after the colon and prefixed with a \sphinxtitleref{\&} sign

\item {} 
\sphinxAtStartPar
the \sphinxtitleref{\&} prefix sign is used to reference the sample’s path in different places of the same configuration file

\item {} 
\sphinxAtStartPar
example use:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{paths}\PYG{p}{:}
  \PYG{n}{mysample}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{mysample} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{mysample}\PYG{o}{.}\PYG{n}{bam}
  \PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mf}{123.}\PYG{n}{bam}
  \PYG{n}{potato95}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{potato95} \PYG{o}{/}\PYG{n}{home}\PYG{o}{/}\PYG{n}{john}\PYG{o}{/}\PYG{n}{bam\PYGZus{}files}\PYG{o}{/}\PYG{n}{potato95}\PYG{o}{.}\PYG{n}{bam}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{bed\_path}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
path to bed file if using varying window sizes

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{fasta\_path}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
path to fasta file

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{output\_path}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
path of output files (references, plots, CNVs) to the \sphinxstylestrong{pbgl\sphinxhyphen{}cnvkit/output} directory

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{references}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
references to use for making comparisons

\item {} 
\sphinxAtStartPar
a reference can be built from multiple “normal” files, which are in turn listed under \sphinxtitleref{files\_for\_ref}

\item {} 
\sphinxAtStartPar
an output reference name needs to be defined

\item {} 
\sphinxAtStartPar
example use:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{references}\PYG{p}{:}
  \PYG{n}{first\PYGZus{}reference}\PYG{p}{:}
    \PYG{n}{output\PYGZus{}ref}\PYG{p}{:} \PYG{o}{\PYGZam{}}\PYG{n}{first\PYGZus{}reference} \PYG{n}{my\PYGZus{}first\PYGZus{}reference}\PYG{o}{.}\PYG{n}{cnn}
    \PYG{n}{files\PYGZus{}for\PYGZus{}ref}\PYG{p}{:}
      \PYG{o}{\PYGZhy{}} \PYG{o}{*}\PYG{n}{first\PYGZus{}bam}
      \PYG{o}{\PYGZhy{}} \PYG{o}{*}\PYG{n}{second\PYGZus{}bam}
      \PYG{o}{\PYGZhy{}} \PYG{o}{*}\PYG{n}{third\PYGZus{}bam}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{comparisons}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
comparison names with respective reference and mutant samples per comparison

\item {} 
\sphinxAtStartPar
each comparison can be named as desired

\item {} 
\sphinxAtStartPar
the sample names to be used as \sphinxtitleref{control} and \sphinxtitleref{mutant} need to be prefixed by a \sphinxtitleref{*} sign

\item {} 
\sphinxAtStartPar
the \sphinxtitleref{*} prefixed sign is used to extract the sample’s path defined in the \sphinxtitleref{paths} section

\item {} 
\sphinxAtStartPar
example:

\end{itemize}

\end{itemize}

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{comparisons}\PYG{p}{:}
  \PYG{n}{variety}\PYG{o}{\PYGZhy{}}\PYG{n}{x}\PYG{p}{:}
    \PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{1}\PYG{p}{:}
      \PYG{n}{reference}\PYG{p}{:} \PYG{o}{*}\PYG{n}{reference}\PYG{o}{\PYGZhy{}}\PYG{n}{one}
      \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{potato95}
    \PYG{n}{a}\PYG{o}{\PYGZhy{}}\PYG{n}{different}\PYG{o}{\PYGZhy{}}\PYG{n}{comparison}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{278}\PYG{n}{asd}\PYG{p}{:}
      \PYG{n}{reference}\PYG{p}{:} \PYG{o}{*}\PYG{n}{another}\PYG{o}{\PYGZhy{}}\PYG{n}{reference}
      \PYG{n}{mutant}\PYG{p}{:} \PYG{o}{*}\PYG{n}{XYZ}\PYG{o}{\PYGZhy{}}\PYG{l+m+mi}{123}
\end{sphinxVerbatim}
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxtitleref{chromosomes}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
list of chromosome names to analyze

\item {} 
\sphinxAtStartPar
chromosome names can be extracted from a bam file’s header

\end{itemize}

\item {} 
\sphinxAtStartPar
\sphinxtitleref{cores}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
a digit, specifying the number of cores to parallelize the workflow

\end{itemize}

\end{itemize}


\section{Running the cnvkit\sphinxhyphen{}analysis Jupyter Notebook}
\label{\detokenize{index:running-the-cnvkit-analysis-jupyter-notebook}}
\begin{sphinxadmonition}{note}{Note:}
\sphinxAtStartPar
It is recommended to duplicate the \sphinxstylestrong{cnvkit\sphinxhyphen{}analysis.ipynb} notebook and then renaming the copy before doing any edits to the notebook.
\end{sphinxadmonition}

\sphinxAtStartPar
Click on \sphinxstylestrong{cnvkit\sphinxhyphen{}analysis.ipynb} and a new tab will open the notebook.

\sphinxAtStartPar
The notebook contains cells that are populated by text or code. Information about each command is provided in the notebook to guide the user. It consists of four parts:
\begin{enumerate}
\sphinxsetlistlabels{\arabic}{enumi}{enumii}{}{.}%
\item {} 
\sphinxAtStartPar
Setup and Configuration File Extraction

\item {} 
\sphinxAtStartPar
Reference Creation

\item {} 
\sphinxAtStartPar
Comparisons

\item {} 
\sphinxAtStartPar
Plotting

\end{enumerate}


\subsection{Setup and Configuration File Extraction}
\label{\detokenize{index:setup-and-configuration-file-extraction}}
\sphinxAtStartPar
A configuration file config.cnvkit.yml in the config/ directory is provided for specifying file paths, references to build, comparisons to analyze, chromosomes to plot, and cores for parallelization.

\sphinxAtStartPar
All the analyses are done by extracting parameters from the configuration file, looping with Python, and running bash system commands through Python’s os library.


\subsection{Reference Creation}
\label{\detokenize{index:reference-creation}}
\sphinxAtStartPar
Compiling a copy\sphinxhyphen{}number reference from given files or directory (containing normal samples). The reference can be constructed from zero, one or multiple control samples. If given a reference genome, also calculate the GC content and repeat\sphinxhyphen{}masked proportion of each region. Files needed:
\begin{itemize}
\item {} 
\sphinxAtStartPar
bam files of normal/control sample(s)

\item {} 
\sphinxAtStartPar
fasta file

\item {} 
\sphinxAtStartPar
bed file with target regions

\end{itemize}

\sphinxAtStartPar
There are two ways to run the command:


\subsubsection{Option 1}
\label{\detokenize{index:option-1}}
\sphinxAtStartPar
Using wildcard * to specify all normal/control files to use for reference building.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{cnvkit}\PYG{o}{/}\PYG{n}{cnvkit}\PYG{o}{.}\PYG{n}{py} \PYG{n}{batch} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{normal} \PYG{n}{normalFile}\PYG{o}{*}\PYG{o}{.}\PYG{n}{bam} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{output}\PYG{o}{\PYGZhy{}}\PYG{n}{reference} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{nameOfReferenceToCreate}\PYG{o}{.}\PYG{n}{cnn} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{fasta} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{fastaFile}\PYG{o}{.}\PYG{n}{fna} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{targets} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{bedFile}\PYG{o}{.}\PYG{n}{bed} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{output}\PYG{o}{\PYGZhy{}}\PYG{n+nb}{dir} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{numberOfCoresToUseForParallelization}
\end{sphinxVerbatim}


\subsubsection{Option 2}
\label{\detokenize{index:option-2}}
\sphinxAtStartPar
Listing each normal/control file separately if wildcard cannot be applied.

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{cnvkit}\PYG{o}{/}\PYG{n}{cnvkit}\PYG{o}{.}\PYG{n}{py} \PYG{n}{batch} \PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{normal} \PYG{n}{normalFile1}\PYG{o}{.}\PYG{n}{bam} \PYG{n}{normalFile2}\PYG{o}{.}\PYG{n}{bam} \PYG{n}{normalFileN}\PYG{o}{.}\PYG{n}{bam} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{output}\PYG{o}{\PYGZhy{}}\PYG{n}{reference} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{nameOfReferenceToCreate}\PYG{o}{.}\PYG{n}{cnn} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{fasta} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{fastaFile}\PYG{o}{.}\PYG{n}{fna} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{targets} \PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{bedFile}\PYG{o}{.}\PYG{n}{bed} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{o}{\PYGZhy{}}\PYG{n}{output}\PYG{o}{\PYGZhy{}}\PYG{n+nb}{dir} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{numberOfCoresToUseForParallelization}
\end{sphinxVerbatim}


\subsection{Comparisons}
\label{\detokenize{index:comparisons}}
\sphinxAtStartPar
Using a reference for calculating coverage in the given regions from BAM read depths. Command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{cnvkit}\PYG{o}{/}\PYG{n}{cnvkit}\PYG{o}{.}\PYG{n}{py} \PYG{n}{batch} \PYG{n}{mutantFile}\PYG{o}{.}\PYG{n}{bam} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{r} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{reference}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{referenceFile}\PYG{o}{.}\PYG{n}{cnn} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{d} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}
\PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{numberOfCoresToUseForParallelization}
\end{sphinxVerbatim}


\subsection{Plotting}
\label{\detokenize{index:plotting}}
\sphinxAtStartPar
Plot bin\sphinxhyphen{}level log2 coverages and segmentation calls together. Without any further arguments, this plots the genome\sphinxhyphen{}wide copy number in a form familiar to those who have used array comparative genomic hybridization (aCGH). The options \textendash{}chromosome or \sphinxhyphen{}c focuses the plot on the specified region. Command:

\begin{sphinxVerbatim}[commandchars=\\\{\}]
\PYG{n}{cnvkit}\PYG{o}{/}\PYG{n}{cnvkit}\PYG{o}{.}\PYG{n}{py} \PYG{n}{scatter} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{mutantFileName}\PYG{o}{.}\PYG{n}{cnr} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{s} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{mutantFileName}\PYG{o}{.}\PYG{n}{cns} \PYGZbs{}
\PYG{o}{\PYGZhy{}}\PYG{n}{c} \PYG{n}{chromosomeName}
\PYG{o}{\PYGZhy{}}\PYG{n}{o} \PYG{o}{/}\PYG{n}{output}\PYG{o}{/}\PYG{n}{path}\PYG{o}{/}\PYG{n}{nameOfPlot}\PYG{o}{.}\PYG{n}{png}
\PYG{o}{\PYGZhy{}}\PYG{n}{p} \PYG{n}{numberOfCoresToUseForParallelization}
\end{sphinxVerbatim}

\sphinxAtStartPar
To run a cell, click on the corresponding cell and press \sphinxstylestrong{Ctrl + Enter} or \sphinxstylestrong{Shift + Enter}.


\section{References}
\label{\detokenize{index:references}}
\sphinxAtStartPar
\sphinxstylestrong{BMC Bioinformatics Publication}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
Talevich, E., Shain, A. H., Botton, T., \& Bastian, B. C. (2014). CNVkit: Genome\sphinxhyphen{}wide copy number detection and visualization from targeted sequencing. PLOS Computational Biology 12(4): e1004873. doi: 10.1371/journal.pcbi.1004873

\end{itemize}

\sphinxAtStartPar
\sphinxstylestrong{GitHub repositories}:
\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/etal/cnvkit.git}{etal/cnvkit}

\item {} 
\sphinxAtStartPar
\sphinxhref{https://github.com/amora197/cnvkit.git}{amora197/cnvkit}

\end{itemize}



\renewcommand{\indexname}{Index}
\printindex
\end{document}